### PR TITLE
refactor(connector): [Stax] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by shivansh-bhatnagar18**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR solves the Currency Unit Conversion for STAX Connector and fixes issue #2246 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We need to create a payment using Stax and test the currency unit from connector response and connector dashboard.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hey @shivansh-bhatnagar18 ,
Have you addressed lints thrown by `cargo` `clippy`?

**Comment by shivansh-bhatnagar18** [comment]
No, will do it ASAP

**Comment by deepanshu-iiitu** [comment]
Hi @shivansh-bhatnagar18 
Please address the CI checks compilation errors.

Also use the following commands before committing your code

```
cargo +nightly fmt
cargo check
cargo run --features openapi -- generate-openapi-spec
```

File: crates/router/src/connector/stax/transformers.rs [hunk 1]
```diff
@@ -11,6 +11,36 @@ use crate::{
...
+            f64,
+            T,
+        ),
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount,
+            router_data: item,
+        })
+    }
+}
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:42]:
```suggestion
    type Error = error_stack::Report<errors::ConnectorError>;
    fn try_from(
        (currency_unit, currency, amount, item): (
            &types::api::CurrencyUnit,
            types::storage::enums::Currency,
            f64,
            T,
        ),
    ) -> Result<Self, Self::Error> {
    let amount = utils::get_amount_as_f64(currency_unit, amount, currency)?;
        Ok(Self {
            amount,
            router_data: item,
        })
    }
}
```

File: crates/router/src/connector/stax/transformers.rs [hunk 2]
```diff
@@ -26,21 +56,23 @@ pub struct StaxPaymentsRequest {
...
+    ) -> Result<Self, Self::Error> {
+        if item.router_data.request.currency != enums::Currency::USD {
             Err(errors::ConnectorError::NotSupported {
-                message: item.request.currency.to_string(),
+                message: item.router_data.request.currency.to_string(),
                 connector: "Stax",
             })?
         }
-        let total = utils::to_currency_base_unit_asf64(item.request.amount, item.request.currency)?;
+        let total = item.amount as f64;
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:70]:
```suggestion
        let total = item.amount;
```

File: crates/router/src/connector/stax/transformers.rs [hunk 3]
```diff
@@ -347,13 +379,12 @@ pub struct StaxCaptureRequest {
...
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
-        let total = utils::to_currency_base_unit_asf64(
-            item.request.amount_to_capture,
-            item.request.currency,
-        )?;
+    fn try_from(
+        item: &StaxRouterData<&types::PaymentsCaptureRouterData>,
+    ) -> Result<Self, Self::Error> {
+        let total = item.amount as f64;
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:387]:
```suggestion
        let total = item.amount;
```

File: crates/router/src/connector/stax/transformers.rs [hunk 4]
```diff
@@ -365,14 +396,11 @@ pub struct StaxRefundRequest {
...
+impl<F> TryFrom<&StaxRouterData<&types::RefundsRouterData<F>>> for StaxRefundRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+    fn try_from(item: &StaxRouterData<&types::RefundsRouterData<F>>) -> Result<Self, Self::Error> {
         Ok(Self {
-            total: utils::to_currency_base_unit_asf64(
-                item.request.refund_amount,
-                item.request.currency,
-            )?,
+            total: item.amount as f64,
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:403]:
```suggestion
            total: item.amount,
```

**Review by AkshayaFoiger — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/stax/transformers.rs [hunk 5]
```diff
@@ -11,6 +11,37 @@ use crate::{
...
+        i64,
+        T,
+    )> for StaxRouterData<T>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        (currency_unit, currency, amount, item): (
+            &types::api::CurrencyUnit,
+            types::storage::enums::Currency,
+            f64,
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:33]:
```suggestion
            i64,
```

File: crates/router/src/connector/stax/transformers.rs [hunk 6]
```diff
@@ -11,6 +11,37 @@ use crate::{
     types::{self, api, storage::enums},
 };
 
+#[derive(Debug, Serialize)]
+pub struct StaxRouterData<T> {
+    pub amount: i64,
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/stax/transformers.rs:16]: Let amount be f64
```suggestion
    pub amount: f64,
```

**Review by AkshayaFoiger — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/stax/transformers.rs (hunk 6 above)
Comment (shivansh-bhatnagar18) [reply crates/router/src/connector/stax/transformers.rs:16]: Ohh thank you for pointing out my changes

**Review by AkshayaFoiger — APPROVED** [review]

**Review by swangi-kumari — APPROVED** [review]
LGTM!

**Comment by shivansh-bhatnagar18** [comment]
@deepanshu-iiitu @SamraatBansal @srujanchikke  Can you also please review my changes and merge my pull request?

**Comment by AkshayaFoiger** [comment]
Hello @shivansh-bhatnagar18 ,
your PR is approved. It will get merged by Monday.

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @shivansh-bhatnagar18!

**Comment by deepanshu-iiitu** [comment]
Hey @shivansh-bhatnagar18  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**